### PR TITLE
Console: Interactive CLI improvements

### PR DESCRIPTION
**Pull request by medikoo**
- Support joint Console & Dashboard integrations
- Do not list not supported by Console services when Console integration is to be setup
- Fix discovered logic bugs
- Introduce additional `isConsoleEnabled` property for `initialContext`

### CODE REVIEW

File: lib/cli/interactive-setup/service.js [hunk 1]
```diff
@@ -32,12 +32,15 @@ const initializeProjectChoices = [
...
-const projectTypeChoice = async (stepHistory) => {
+const projectTypeChoice = async ({ stepHistory, options }) => {
   const projectType = await promptWithHistory({
     message: 'What do you want to make?',
     type: 'list',
     name: 'projectType',
-    choices: initializeProjectChoices,
+    choices: initializeProjectChoices.filter(
+      // Temporarily filter out Python and SQS projects when console support is intended
+      ({ value }) => !options.console || (!value.includes('python') && !value.includes('sqs'))
```
Comment (pgrzesik) [inline lib/cli/interactive-setup/service.js:42]: what about the starter templates that don't have any events? Are these supported by console? Last time I was checking these functions' invocations weren't showing up (only API)

File: lib/cli/interactive-setup/dashboard-login.js [hunk 2]
```diff
@@ -17,9 +17,7 @@ const loginOrRegisterQuestion = async (stepHistory, isConsole) =>
 
 const steps = {
   loginOrRegister: async (context) => {
-    const isConsole =
```
Comment (pgrzesik) [inline lib/cli/interactive-setup/dashboard-login.js:20]: Thanks for addressing that - I didn't recognize here that its possible to support both integrations

File: lib/cli/interactive-setup/utils.js [hunk 3]
```diff
@@ -54,7 +54,12 @@ module.exports = {
       isInServiceContext: Boolean(serviceDir),
       isLoggedIntoDashboard: isAuthenticated(),
       hasLocalAwsCredentials: hasLocalCredentials(),
-      isDashboardEnabled: Boolean(configuration && configuration.org && configuration.app),
+      isDashboardEnabled: Boolean(
+        configuration &&
+          configuration.org &&
+          configuration.app &&
+          (!configuration.console || configuration.dashboard)
```
Comment (pgrzesik) [inline lib/cli/interactive-setup/utils.js:62]: Is that check needed? Isn't having `app` defined is enough here?

**Review by pgrzesik — COMMENTED** [review]
Looks good overall, I only have a few minor questions

File: lib/cli/interactive-setup/service.js (hunk 1 above)
Comment (medikoo) [reply lib/cli/interactive-setup/service.js:42]: Technically all Node.js lambda invocations are reported. If they do not appear in Console, I'd take it's an issue that either was already handled or will be handled shortly

File: lib/cli/interactive-setup/utils.js (hunk 3 above)
Comment (medikoo) [reply lib/cli/interactive-setup/utils.js:62]: Not exactly, as `configuration.console` disables _dashboard_ support, unless `dashboard: true` is set.

**Review by pgrzesik — APPROVED** [review]

**Pull request merged**